### PR TITLE
Minor fixups for future NumPy 2 compatibility

### PR DESCRIPTION
**Pull request by seberg**
These are some small fixes to be compatible with NumPy 2 python changes, as pointed out by the `ruff` with the `"NPY201"` rule-set.

I am not really happy with the `_NUMPY_SCTYPES` (reaching into what is now private, but figured that others will do so also for a while; feels like we should add a better way to do this in NumPy before removing it).
Listing the full set is also a bit ugly/convoluted, but happy to do so instead.

(I was hoping to get a bit further with testing against the NumPy 2rc, but unfortunately the `numba` dependency makes that at least difficult.)

### CODE REVIEW

File: python/cudf/cudf/tests/test_api_types.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def test_is_categorical_dtype(obj, expect):
         (np.float64(), True),
         (np.complex128(), True),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:154]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 2]
```diff
@@ -256,7 +256,7 @@ def test_is_numeric_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:259]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 3]
```diff
@@ -247,7 +247,7 @@ def test_is_numeric_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:250]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 4]
```diff
@@ -275,7 +275,7 @@ def test_is_numeric_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:278]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 5]
```diff
@@ -352,7 +352,7 @@ def test_is_integer_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:355]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 6]
```diff
@@ -361,7 +361,7 @@ def test_is_integer_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:364]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 7]
```diff
@@ -380,7 +380,7 @@ def test_is_integer_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:383]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 8]
```diff
@@ -458,7 +458,7 @@ def test_is_integer(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, True),
-        (np.unicode_, True),
+        (np.str_, True),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:461]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 9]
```diff
@@ -467,7 +467,7 @@ def test_is_integer(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), True),
-        (np.unicode_(), True),
+        (np.str_(), True),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:470]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 10]
```diff
@@ -486,7 +486,7 @@ def test_is_integer(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), True),
-        (np.array([], dtype=np.unicode_), True),
+        (np.array([], dtype=np.str_), True),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:489]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 11]
```diff
@@ -577,7 +577,7 @@ def test_is_string_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:580]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 12]
```diff
@@ -586,7 +586,7 @@ def test_is_string_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:589]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 13]
```diff
@@ -605,7 +605,7 @@ def test_is_string_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:608]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 14]
```diff
@@ -682,7 +682,7 @@ def test_is_datetime_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:685]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 15]
```diff
@@ -691,7 +691,7 @@ def test_is_datetime_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:694]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 16]
```diff
@@ -710,7 +710,7 @@ def test_is_datetime_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:713]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 17]
```diff
@@ -787,7 +787,7 @@ def test_is_list_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:790]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 18]
```diff
@@ -796,7 +796,7 @@ def test_is_list_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:799]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 19]
```diff
@@ -815,7 +815,7 @@ def test_is_list_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:818]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 20]
```diff
@@ -895,7 +895,7 @@ def test_is_struct_dtype(obj, expect):
         (np.float64, False),
         (np.complex128, False),
         (np.str_, False),
-        (np.unicode_, False),
+        (np.str_, False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:898]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 21]
```diff
@@ -904,7 +904,7 @@ def test_is_struct_dtype(obj, expect):
         (np.float64(), False),
         (np.complex128(), False),
         (np.str_(), False),
-        (np.unicode_(), False),
+        (np.str_(), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:907]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 22]
```diff
@@ -923,7 +923,7 @@ def test_is_struct_dtype(obj, expect):
         (np.array([], dtype=np.float64), False),
         (np.array([], dtype=np.complex128), False),
         (np.array([], dtype=np.str_), False),
-        (np.array([], dtype=np.unicode_), False),
+        (np.array([], dtype=np.str_), False),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:926]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 23]
```diff
@@ -1004,7 +1004,7 @@ def test_is_decimal_dtype(obj, expect):
         np.float64,
         np.complex128,
         np.str_,
-        np.unicode_,
+        np.str_,
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:1007]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 24]
```diff
@@ -1013,7 +1013,7 @@ def test_is_decimal_dtype(obj, expect):
         np.float64(),
         np.complex128(),
         np.str_(),
-        np.unicode_(),
+        np.str_(),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:1016]:
```suggestion
```

File: python/cudf/cudf/tests/test_api_types.py [hunk 25]
```diff
@@ -1032,7 +1032,7 @@ def test_is_decimal_dtype(obj, expect):
         np.array([], dtype=np.float64),
         np.array([], dtype=np.complex128),
         np.array([], dtype=np.str_),
-        np.array([], dtype=np.unicode_),
+        np.array([], dtype=np.str_),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_api_types.py:1035]:
```suggestion
```

**Review by galipremsagar — CHANGES_REQUESTED** [review]
We could rather drop the `np.unicode_` from our code instead of duplicating with `np.str_` because all tests have this covered.

**Comment by seberg** [comment]
> We could rather drop the np.unicode_ from our code instead of duplicating with np.str_ because all tests have this covered.

Yeah good point, should have noticed that it was just duplicating the tests.

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**